engine/runtime/src/main/java/org/enso/interpreter/node/expression/builtin/number/decimal/RoundNode.java

```diff
@@ -0,0 +1,103 @@
+package org.enso.interpreter.node.expression.builtin.number.decimal;
+
+import com.oracle.truffle.api.CompilerDirectives;
+import com.oracle.truffle.api.CompilerDirectives.TruffleBoundary;
+import com.oracle.truffle.api.nodes.Node;
+import com.oracle.truffle.api.profiles.CountingConditionProfile;
+import java.math.BigDecimal;
+import java.math.BigInteger;
+import org.enso.interpreter.dsl.BuiltinMethod;
+import org.enso.interpreter.node.expression.builtin.number.utils.BigIntegerOps;
+import org.enso.interpreter.runtime.EnsoContext;
+import org.enso.interpreter.runtime.builtin.Builtins;
+import org.enso.interpreter.runtime.data.text.Text;
+import org.enso.interpreter.runtime.error.PanicException;
+import org.enso.interpreter.runtime.number.EnsoBigInteger;
+
+@BuiltinMethod(
+        type = "Decimal",
+        name = "round",
+        description = "Decimal ceiling, converting to a small or big integer depending on size.")
+public class RoundNode extends Node {
+    private final CountingConditionProfile fitsProfile = CountingConditionProfile.create();
+
+    /** Minimum value for the `decimal_places` parameter to `roundDouble`. */
+    private static final double ROUND_MIN_DECIMAL_PLACES = -15;
+
+    /** Maximum value for the `decimal_places` parameter to `roundDouble`. */
+    private static final double ROUND_MAX_DECIMAL_PLACES = 15;
+
+    /** Minimum value for the `n` parameter to `roundDouble`. */
+    private static final double ROUND_MIN_DOUBLE = -99999999999999.0;
+
+    /** Minimum value for the `n` parameter to `roundDouble`. */
+    private static final double ROUND_MAX_DOUBLE = 99999999999999.0;
+
+    Object execute(double n, long decimalPlaces, boolean useBankers) {
```

Comment:
Please _profile_ value of `decimalPlaces` with [PrimitiveValueProfile](https://www.graalvm.org/truffle/javadoc/com/oracle/truffle/api/profiles/PrimitiveValueProfile.html). If the value of `decimalPlaces` is always the same then the compiler can take it as constant. Most of the code in this method can be precomputed in compilation time. I believe the result is going to be very visible in the benchmark.

Reply:
Done